client/users/user-profile-seasons.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+import { appendToMultimap } from '../../common/data-structures/maps'
+import { LadderPlayer } from '../../common/ladder/ladder'
+import { SeasonId } from '../../common/matchmaking'
+import { SbUser } from '../../common/users/sb-user'
+import { LoadingDotsArea } from '../progress/dots'
+import { useAppDispatch, useAppSelector } from '../redux-hooks'
+import { colorError, colorTextFaint, colorTextPrimary } from '../styles/colors'
+import { headline6, subtitle1 } from '../styles/typography'
+import { getUserRankingHistory } from './action-creators'
+import { UserRankDisplay } from './user-rank-display'
+
+const Container = styled.div`
+  padding: 0 24px;
+
+  display: flex;
+  flex-direction: column;
+  gap: 48px;
+`
+
+const NoResults = styled.div`
+  ${subtitle1};
+
+  color: ${colorTextFaint};
+`
+
+const ErrorText = styled.div`
+  ${subtitle1};
+
+  color: ${colorError};
+`
+
+export function UserProfileSeasons({ user }: { user: SbUser }) {
+  const { t } = useTranslation()
+  const dispatch = useAppDispatch()
+
+  const [ranks, setRanks] = useState<LadderPlayer[]>()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [error, setError] = useState<Error>()
+  const abortControllerRef = useRef<AbortController>()
+
+  useEffect(() => {
+    setIsLoading(true)
+
+    abortControllerRef.current?.abort()
+    abortControllerRef.current = new AbortController()
+
+    dispatch(
+      getUserRankingHistory(user.id, {
+        signal: abortControllerRef.current.signal,
+        onSuccess: data => {
+          setIsLoading(false)
+          setRanks(data.history)
+        },
+        onError: err => {
+          setIsLoading(false)
+          setError(err)
+        },
+      }),
+    )
+  }, [dispatch, user.id])
+
+  useEffect(() => {
+    return () => abortControllerRef.current?.abort()
```

Comment:
Think this will cause issues, it should either be part of the previous effect or not included at all